LoopKit/DeviceManager/DeviceManager.swift

```diff
@@ -8,20 +8,25 @@
 import Foundation
 import UserNotifications
 
-public protocol DeviceManagerDelegate {
+public protocol DeviceManagerDelegate: UserAlertHandler {
+    
+    #if !USE_NEW_ALERT_FACILITY
     func scheduleNotification(for manager: DeviceManager,
                               identifier: String,
                               content: UNNotificationContent,
                               trigger: UNNotificationTrigger?)
 
     func clearNotification(for manager: DeviceManager, identifier: String)
-    
+
     func removeNotificationRequests(for manager: DeviceManager, identifiers: [String])
-    
+    #endif
     func deviceManager(_ manager: DeviceManager, logEventForDeviceIdentifier deviceIdentifier: String?, type: DeviceLogEntryType, message: String, completion: ((Error?) -> Void)?)
+
+    var alertHandler: UserAlertHandler? { get set }
```

Comment:
I think keeping alert handling a direct responsibility of DeviceManagerDelegate is fine, especially because it's just 3 methods, and that's how it is now.  If you want to keep it separate (maybe you want clients to be able to provide separate objects to a) handle alerts and b) handle general device delegate responsibilities), then I think it should be a separate delegate, and not nested like this.  It's also seems odd to have `DeviceManagerDelegate` conform to `UserAlertHandler` and also serve up a UserAlertHandler.

Reply:
Yeah, it is odd, but it is the way I designed it to "pass along" the calls up the stack.  The alternative would be to pass in the `UserAlertManager` all the way down into the device managers -- which I wanted to avoid, since we already had the Delegates wired up.  Perhaps worth a quick live discussion?

And, yes, it was my intention to keep separate the "alert handling" responsibility instead of continuing to grow the responsibility of `DeviceDataManager` and the `DeviceManagerDelegate`.

Reply:
As per our discussion, I do think it's fine to separate the Alert handling out of DeviceManagerDelegate. But having DeviceManagerDelegate conform to `UserAlertHandler` and also vend a `UserAlertHandler` instance seems redundant.

Reply:
`DeviceManagerDelegate` does conform to `UserAlertHandler` -- see line 11.  I'll take out the vending of the instance (yes, it is a little redundant).